homeassistant/components/nextbus/sensor.py

```diff
@@ -70,23 +72,28 @@ async def async_setup_entry(
     async_add_entities: AddEntitiesCallback,
 ) -> None:
     """Load values from configuration and initialize the platform."""
-    client = NextBusClient(output_format="json")
-
     _LOGGER.debug(config.data)
-
-    sensor = NextBusDepartureSensor(
-        client,
-        config.unique_id,
-        config.data[CONF_AGENCY],
-        config.data[CONF_ROUTE],
-        config.data[CONF_STOP],
-        config.data.get(CONF_NAME) or config.title,
+    entry_agency = config.data[CONF_AGENCY]
+
+    coordinator: NextBusDataUpdateCoordinator = hass.data[DOMAIN].get(entry_agency)
```

Comment:
We shouldn't use `dict.get(key)` if we know that the key is in the dict. Use `dict[key]` instead.